azure_pipeline/src/utils/api.ts

```diff
@@ -0,0 +1,40 @@
+import { http } from "./http";
+import { AxiosRequestConfig } from "axios";
+
+export type UserConfig = {
+    properties: {
+        rootFile: string;
+        files?: NginxConfigurationFile[];
+        package?: NginxConfigurationPackage;
+    };
+};
+
+export type NginxConfigurationFile = {
+    virtualPath: string;
+    content: string;
+}
+
+export type NginxConfigurationPackage = {
+    data: string;
+}
+
+export type ResourceInfo = {
+    subscriptionId: string;
+    resourceGroupName: string;
+    deploymentName: string;
+    apiVersion: string;
+}
+
+
+export const updateUserConfig = async (
```

Comment:
is it possible to add a unit test for this?

Reply:
Indeed! I have added some unit test for this at `update-user-config.test.ts` . Any advice is muchly appreciated!

Reply:
awesome, looks good to me!